purrfect-match/src/ContactPage.tsx

```diff
@@ -63,6 +75,16 @@ export class ContactPage extends React.Component<ContactPageProps, ContactPageSt
                     modalMessage: "Submitted request successfully!",
                     modalOpen: true
                 });
+             
+                if (this.validateEmail){
+                    this.setState({
+                        modalMessage: "Please submit a valid email!",
+                        modalOpen: true
+                });
```

Comment:
Have you tested this? Why are you validating the email after submitting a contact request instead of before? 

Also, all the other validation errors appear with the field instead of in a modal. Maybe display this message there, and only use a modal to show errors returned by the backend?